data/systemd/snapd.unshare.service.in

```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=Apply rshared mount to @SNAP_MOUNT_DIR@ directory
+RequiresMountsFor=/snap
+
+[Service]
+Type=oneshot
+ExecStart=/bin/mkdir -p @SNAP_MOUNT_DIR@
+ExecStart=/bin/mount -o rbind @SNAP_MOUNT_DIR@ @SNAP_MOUNT_DIR@
+ExecStart=/bin/mount --make-rshared @SNAP_MOUNT_DIR@
```

Comment:
This can become `--make-shared` as well. The sharing will be inherited by the mouted snaps.